server/src/main/java/org/opensearch/cluster/health/ClusterStateHealth.java

```diff
@@ -162,6 +200,16 @@ public ClusterStateHealth(
         boolean isIndexOrShardLevelHealthRequired = healthLevel == ClusterHealthRequest.Level.INDICES
             || healthLevel == ClusterHealthRequest.Level.SHARDS;
 
+        boolean hasSearchOnlyIndices = false;
```

Comment:
pull this to a private function if repeated pls.

Reply:
I have updated this in my latest commit, please check.